lab_1/main.py

```diff
@@ -87,16 +136,27 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
     right_n = 3
     --> [['man', 'is'], ['dog, 'cat']]
     """
-    pass
-
+    if not isinstance(tokens, list) or not isinstance(word, str) \
+            or not isinstance(left_n, int) or not isinstance(right_n, int):
+        return []
+    concord = get_concordance(tokens, word, left_n, right_n)
+    adj_words = []
+    for element in concord:
+        if left_n > 0 and right_n > 0:
+            adj_words.append([element[0], element[-1]])
+        elif left_n > 0 and not right_n > 0:
+            adj_words.append([element[0]])
+        elif right_n > 0 and not left_n > 0:
+            adj_words.append([element[-1]])
+    return adj_words
 
 def read_from_file(path_to_file: str) -> str:
     """
     Opens the file and reads its content
     :return: the initial text in string format
     """
-    with open(path_to_file, 'r', encoding='utf-8') as fs:
-        data = fs.read()
+    with open(path_to_file, 'r', encoding='utf-8') as file:
```

Comment:
Usually, it is a bad practice to use Python built-ins like `str, list, file` as variable names.